pkg/controller/actuator_reconcile.go

```diff
@@ -110,6 +110,13 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, network *extens
 			networkConfig.IPv4.Mode = (*calicov1alpha1.IPv4PoolMode)(pointer.String(string(calicov1alpha1.Never)))
 			networkConfig.Backend = (*calicov1alpha1.Backend)(pointer.String(string(calicov1alpha1.None)))
 		}
+		if networkConfig.Overlay != nil && networkConfig.Overlay.CreatePodRoutes != nil {
+			if *networkConfig.Overlay.CreatePodRoutes {
+				networkConfig.Backend = (*calicov1alpha1.Backend)(pointer.String(string(calicov1alpha1.Bird)))
+			} else {
+				networkConfig.Backend = (*calicov1alpha1.Backend)(pointer.String(string(calicov1alpha1.None)))
+			}
+		}
```

Comment:
Here it is possible to disable `bird` even if `Overlay` is enabled. 
Is that a valid configuration?
The logic in the related Cilium change seems to be different.

Reply:
Yes, indeed. What do you propose?

Reply:
Maybe something like that:

```
if networkConfig.Overlay != nil && !networkConfig.Overlay.Enabled && networkConfig.Overlay.CreatePodRoutes != nil && networkConfig.Overlay.CreatePodRoutes{
networkConfig.Backend = (*calicov1alpha1.Backend)(pointer.String(string(calicov1alpha1.Bird)))
}
```

Reply:
@DockToFuture @axel7born Is this how you expected it?